docs/mirror.md

```diff
@@ -0,0 +1,85 @@
+
+## How does it work?
+
+The first time you request an image from your local registry mirror, it pulls the image from the public Docker registry and stores it locally before handing it back to you. On subsequent requests, the local registry mirror is able to serve the image from its own storage.
+
+## What if the content changes on the Hub?
+
+When a pull is attempted with a tag, the Registry will check the remote to ensure if it has the latest version of the requested content.  If it doesn't it will fetch the latest content and cache it.
+
+## What about my disk?
+
+In environments with high churn rates, stale data can build up in the cache.  When running as a pull through cache the Registry will periodically remove old content to save disk space.  Subsequent requests for removed content will cause a remote fetch and local re-caching.
+
+To ensure best performance and guarantee correctness the Registry cache should be configured to use the `filesystem` driver for storage.
+
+## Running a Registry as a pull through cache
+
+The easiest way to run a registry as a pull through cache is to run the official Registry pull through cache official image.
+
+### Configuring the cache
+
+To configure a Registry to run as a pull through cache, the following additions are required to the config file.
+
+The `storage` subsection requires the addition of a YAML label.  In this exmple it is set to `&STORAGE`.  Note
+
+```
+storage: &STORAGE
+    cache:
+        blobdescriptor: redis
+    filesystem:
+        rootdirectory: /var/lib/registry_cache
+    maintenance:
+        uploadpurging:
+            enabled: false
+```
+
+The `middleware` subsection requires the following additions.  In order to access private images on the Docker Hub the username and password can be supplied.
+
+```            
+middleware:
+  repository:
+    - name: proxy
+  registry:
+    - name: proxy
+      options:
+        storage: *STORAGE
+        remoteurl: https://registry-1.docker.io
+        username: [username]
+        password: [password]
+```
+
+
+### Building your own Registry pull through cache
+
+Power users may wish to build and run their own Registry as a pull through cache.  In order to do this the following rules must be applied
+
+ * The `noresumabledigest` build tag must be set to true.  The resumable digest mechanism is not used when the Registry is acting as a pull through cache and impacts performance.  
+
+
+## Configuring the Docker daemon
+
+You will need to pass the `--registry-mirror` option to your Docker daemon on startup:
+
+```
+docker --registry-mirror=http://<my-docker-mirror-host> -d
+```
+
+For example, if your mirror is serving on http://10.0.0.2:5000, you would run:
+
+```
+docker --registry-mirror=http://10.0.0.2:5000 -d
```

Comment:
Hmm, wouldn't the docker daemon complain about this example address as it is not https?